src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexPersistence.cs

```diff
@@ -518,18 +533,21 @@ public bool ContainsField(string field)
 
         public void DisposeWriters()
```

Comment:
Is this called concurrently from outside the indexer thread?

Reply:
yes, upon disable or idle clean up.

Reply:
Is it possible that we'll execute index cleanup and trigger indexing work at the same time? I wonder if we can dispose the writer that might be in use. Any chance for that?

Reply:
I have change the code, so now only one thread should modify the index writer.